### PR TITLE
Revamp a bit

### PR DESCRIPTION
**Pull request by Pegase745**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,18 +4,13 @@ language: python
 
 python:
   - "2.7"
-  - "3.5"
-  - "pypy"
+  - "3.6"
```
Comment (tdamsma) [inline .travis.yml]: why not 3.7 also?

File: datatables/datatables.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        self.error = None
+        try:
+            self.run()
+        except Exception as exc:
+            self.error = str(exc)
+
+    def output_result(self):
+        """Output results in the format needed by DataTables."""
+        output = {}
+        output['draw'] = str(int(self.params['draw']))
```
Comment (tdamsma) [inline datatables/datatables.py]: Might as wel include the fix discussed in #98 

```suggestion
        output['draw'] = str(int(self.params.get('draw',1))
```

File: .travis.yml (hunk 1 above)
Comment (tdamsma) [reply .travis.yml]: And come to think of it, any reason to keep supporting legacy python?

Comment (tdamsma) [reply .travis.yml]: I see that 3.7 is not available on Travis... And looking further, actually it is https://github.com/travis-ci/travis-ci/issues/9815. Apparently one must add 

    language: python
    sudo: required
    dist: xenial
    python: "3.7"

**Comment by tdamsma** [comment]
My take at change log from what I just saw

- Convert tests to pytest
- Reorganize code 
- Modernize Flask example
- Modernize pyramid example
- Change and apply different linting style
- Rename `.jinja2` templates to `.html`
- Update to python 3.6
 
And per my suggestion
 - Address #98
 - Drop python 2 support
 -

**Comment by louking** [comment]
Are there not users still "stuck" on 2.7 because of some limited support
from external packages? I know I've been afraid to try to migrate.

File: .travis.yml (hunk 1 above)
Comment (Pegase745) [reply .travis.yml]: I was indeed rejected by Travis, but I'll try your solution. It's best to keep up with the latest versions

**Comment by Pegase745** [comment]
@tdamsma thanks for your inputs. I felt sorry ignoring the project's need for maintenance for so long :) My main concern was to fix our old issue with the Travis tests that failed for no reasons at that time when using an SQLite database. 
I may need help in integrating the current important PRs in the revamp branch, in order to address some of the issues. 
I keep the support for python2 because I see it everyday how much companies are still afraid, or don't have time to migrate due to large base code. For the time being, it doesn't cost too much to keep the compatibility imho

**Comment by tdamsma** [comment]
Are you sure about that last commit, to remove asserts? They are at the core of pytest. Nothing wrong with using asserts for testing, and pytest is built around providing nice error messages for assert statements. Wouldn't be surprised if that is broken now. See https://github.com/fossasia/query-server/issues/332

**Comment by tdamsma** [comment]
About python 2, I guess if is no bother you might as well leave it in, but I haven't touched any python 2 code in over 6 years.

**Comment by tdamsma** [comment]
@Pegase745, You're welcome, keep up the good work!

**Comment by tdamsma** [comment]
@Pegase745, do you think this branch can be released as is?

**Comment by Pegase745** [comment]
will do this tonight then :)

**Pull request merged**